daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -1387,14 +1381,15 @@ trait ElementBaseGrammarMixin
       }
       case LengthKind.Implicit if isSimpleType && primType == PrimType.String =>
         new SpecifiedLengthImplicitCharacters(this, body, this.maxLength.longValue)
-
-      case LengthKind.Implicit if isSimpleType && primType == PrimType.HexBinary =>
-        new SpecifiedLengthImplicit(this, body, this.maxLength.longValue * bitsMultiplier)
       case LengthKind.Implicit
-          if isSimpleType && impliedRepresentation == Representation.Binary =>
+          if isSimpleType &&
+            impliedRepresentation == Representation.Binary &&
+            primType != PrimType.HexBinary =>
         new SpecifiedLengthImplicit(this, body, implicitBinaryLengthInBits)
```

Comment:
This condition feels like it doesn't exclude enough things. For example, this matches all binary types except hex binary. But, for example, integers with length kind implicit use something like `BinaryIntegerKnownLength` primitive/parser, which doesn't rely on SpecifiedLengthImplicit parser. It doesn't necessarily hurt, but it's just unnecessary work and might slow things down, since the BinaryIntegerKnownLengthParser is going to do that work.

I'm wonder if it's just string types that really make use of specified length stuff?

Reply:
So It look like nillable binary types need SpecifiedLengthImplicit., they are the only tests failing when I comment out that chunk of code